pkg/backend/backend_test.go

```diff
@@ -9,15 +9,69 @@ import (
 	"context"
 	"log"
 	"net"
+	"os"
 
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/test/bufconn"
 
 	pb "github.com/opiproject/opi-api/storage/v1alpha1/gen/go"
+	"github.com/opiproject/opi-spdk-bridge/pkg/server"
 )
 
-// TODO: move to a separate (test/server) package to avoid duplication
+// TODO: move test infrastructure code to a separate (test/server) package to avoid duplication
+
+type backendClient struct {
+	pb.NVMfRemoteControllerServiceClient
+	pb.NullDebugServiceClient
+	pb.AioControllerServiceClient
+}
+
+type testEnv struct {
+	opiSpdkServer *Server
+	client        *backendClient
+	ln            net.Listener
+	testSocket    string
+	ctx           context.Context
+	conn          *grpc.ClientConn
+	jsonRPC       server.JSONRPC
+}
+
+func (e *testEnv) Close() {
```

Comment:
can this be in utils ? is it shared ?

Reply:
not yet, if we move it in utils, we will have a circular dependency at the moment. We need to introduce a couple of interfaces to break it.